src/components/ProjectsContainer/ProjectsContainer.jsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import styles from './ProjectsContainer.module.css';
+import { PROJECTS } from '../../constants/constants';
+import { setQuery, updateFilteredItems } from '../../actions';
+import { filterProjects } from '../../utils/filterProjects';
+import { ProjectsList } from '../ProjectsList/ProjectsList';
+import { SearchInput } from '../SearchInput/SearchInput';
+
+const DEBOUNCE_DELAY = 300;
+
+export const ProjectsContainer = () => {
+    const dispatch = useDispatch();
+    const query = useSelector((state) => state.projects.query);
+    const filteredItems = useSelector((state) => state.projects.filteredItems);
+
+    useEffect(() => {
```

Comment:
Usually the input component itself is responsible for debouncing, not its parent. For example parent component can store debounced value and child input component will store its state and update parent state with debounce.